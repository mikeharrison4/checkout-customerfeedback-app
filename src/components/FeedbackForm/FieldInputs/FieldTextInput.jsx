import React from 'react';

const FieldTextInput = ({
  field,
  label,
  className = '',
  form: { isSubmitting, values, errors, touched },
  ...props
}) => {
  const hasErrorOccurred = errors[field.name] && touched[field.name];
  return (
    <div className={`relative w-full ${className}`}>
      <input
        { ...field }
        {...props}
        disabled={isSubmitting}
        className={`bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 ${hasErrorOccurred ? 'border-pink-500' : ''}`}
        autoComplete="off"
      />
      <label
        htmlFor={field.name}
        className={`w-full absolute pointer-events-none left-0 top-1 text-xl cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all ${values[field.name] ? '-top-4 text-xs text-white' : ''}`}
      >
        <span>{label}</span>
        { hasErrorOccurred
          && (
            <span className="absolute text-xs right-0 text-pink-500">
              {errors[field.name]}
            </span>
          )
        }
      </label>
    </div>
  );
};

export default FieldTextInput;
