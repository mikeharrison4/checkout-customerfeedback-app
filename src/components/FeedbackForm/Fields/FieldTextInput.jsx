import React from 'react';
import FieldLabel from './FieldLabel';

const FieldTextInput = ({
  field,
  label,
  className = '',
  form: { isSubmitting, values, errors, touched },
  ...props
}) => {
  const hasErrorOccurred = errors[field.name] && touched[field.name];
  return (
    <div className={`relative w-full ${className}`}>
      <input
        {...field}
        {...props}
        id={field.name}
        aria-labelledby={label}
        disabled={isSubmitting}
        className={`bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 ${hasErrorOccurred ? 'border-pink-500' : ''}`}
      />
      <FieldLabel
        name={field.name}
        label={label}
        hasErrorOccurred={hasErrorOccurred}
        error={errors[field.name]}
        className={`w-full absolute pointer-events-none left-0 cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all ${values[field.name] ? '-top-4 text-xs' : 'text-xl top-2'}`}
      />
    </div>
  );
};

export default FieldTextInput;
