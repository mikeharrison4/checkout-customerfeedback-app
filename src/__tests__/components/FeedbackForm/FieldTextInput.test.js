import React from 'react';
import { shallow } from 'enzyme';
import FieldTextInput from '../../../components/FeedbackForm/Fields/FieldTextInput';

describe('FieldTextInput.jsx', () => {
  let props;
  
  beforeEach(() => {
    props = {
      field: {
        name: 'mockFieldName'
      },
      label: 'mockLabel',
      className: 'mockClassName',
      form: {
        isSubmitting: false,
        values: { mockFieldName: 'mockName' },
        errors: {},
        touched: {}, 
      }
    };
  });
  
  it('should render correctly when no errors are present', () => {
    const renderedComponent = shallow(<FieldTextInput {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render correctly when no className is provided', () => {
    delete props.className;
    const renderedComponent = shallow(<FieldTextInput {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render correctly when field value is empty', () => {
    props.form.values = {};
    const renderedComponent = shallow(<FieldTextInput {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render correctly errors are present but the field has not been touched', () => {
    props.form.errors = { mockFieldName: 'Required' };
    const renderedComponent = shallow(<FieldTextInput {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should show errors when errors are present and the field has been touched', () => {
    props.form.errors = { mockFieldName: 'Required' };
    props.form.touched = { mockFieldName: true };
    const renderedComponent = shallow(<FieldTextInput {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

});