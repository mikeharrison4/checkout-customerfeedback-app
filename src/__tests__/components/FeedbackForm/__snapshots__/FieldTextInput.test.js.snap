// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldTextInput.jsx should render correctly errors are present but the field has not been touched 1`] = `
<div
  className="relative w-full mockClassName"
>
  <input
    aria-labelledby="mockLabel"
    className="bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 "
    disabled={false}
    id="mockFieldName"
    name="mockFieldName"
  />
  <FieldLabel
    className="w-full absolute pointer-events-none left-0 cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all -top-4 text-xs"
    error="Required"
    label="mockLabel"
    name="mockFieldName"
  />
</div>
`;

exports[`FieldTextInput.jsx should render correctly when field value is empty 1`] = `
<div
  className="relative w-full mockClassName"
>
  <input
    aria-labelledby="mockLabel"
    className="bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 "
    disabled={false}
    id="mockFieldName"
    name="mockFieldName"
  />
  <FieldLabel
    className="w-full absolute pointer-events-none left-0 cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all text-xl top-2"
    label="mockLabel"
    name="mockFieldName"
  />
</div>
`;

exports[`FieldTextInput.jsx should render correctly when no className is provided 1`] = `
<div
  className="relative w-full "
>
  <input
    aria-labelledby="mockLabel"
    className="bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 "
    disabled={false}
    id="mockFieldName"
    name="mockFieldName"
  />
  <FieldLabel
    className="w-full absolute pointer-events-none left-0 cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all -top-4 text-xs"
    label="mockLabel"
    name="mockFieldName"
  />
</div>
`;

exports[`FieldTextInput.jsx should render correctly when no errors are present 1`] = `
<div
  className="relative w-full mockClassName"
>
  <input
    aria-labelledby="mockLabel"
    className="bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 "
    disabled={false}
    id="mockFieldName"
    name="mockFieldName"
  />
  <FieldLabel
    className="w-full absolute pointer-events-none left-0 cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all -top-4 text-xs"
    label="mockLabel"
    name="mockFieldName"
  />
</div>
`;

exports[`FieldTextInput.jsx should show errors when errors are present and the field has been touched 1`] = `
<div
  className="relative w-full mockClassName"
>
  <input
    aria-labelledby="mockLabel"
    className="bg-transparent w-full border-b py-2 text-xl text-white focus:outline-none peer focus:invalid:border-pink-500 border-pink-500"
    disabled={false}
    id="mockFieldName"
    name="mockFieldName"
  />
  <FieldLabel
    className="w-full absolute pointer-events-none left-0 cursor-text peer-focus:text-xs peer-focus:-top-4 transition-all -top-4 text-xs"
    error="Required"
    hasErrorOccurred={true}
    label="mockLabel"
    name="mockFieldName"
  />
</div>
`;
